interpluginclient/client.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Package interpluginclient provides a client for interacting with the Mattermost AI plugin
+// from other Mattermost plugins.
+package interpluginclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/plugin"
+	"github.com/pkg/errors"
+)
+
+const (
+	// DefaultTimeout is the default timeout for all requests to the AI plugin
+	DefaultTimeout = 30 * time.Second
```

Comment:
Too low. Long responses and function calls can take way longer than this. I use 10 minutes in the plugin: https://github.com/mattermost/mattermost-plugin-ai/blob/b590abb965a3ff98a0097792958669a4e49280e8/server/plugin.go#L108
(There is also a separate streaming timeout between streaming responses.)

Reply:
Done